pkg/store/storepb/rpc.proto

```diff
@@ -108,18 +108,60 @@ message SeriesRequest {
   google.protobuf.Any hints = 9;
 
   // Query step size in milliseconds.
+  // Deprecated: Use query_hints instead.
   int64 step = 10;
 
   // Range vector selector range in milliseconds.
+  // Deprecated: Use query_hints instead.
   int64 range = 11;
+
+  // Indicates whether the store should evaluate queries
+  // before sending back samples.
+  bool enableQueryPushdown = 12;
```

Comment:
Maybe stupid idea, but why not controling if query pushdown should be send or not by either sending query hints or not sending them?

Reply:
Less params == simpler